### PR TITLE
Correct message for invalid indentation in doc test when is "1 space"

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/doc_test.ex [hunk 1]
```diff
@@ -438,7 +438,8 @@ defmodule ExUnit.DocTest do
       ^stripped_line ->
         :ok
       _ ->
-        raise Error, message: "indentation level mismatch: #{inspect line}, should have been #{indent} spaces"
+        if indent == 1, do: n_spaces = "#{indent} space", else: n_spaces = "#{indent} spaces"
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/doc_test.ex]: Let's take `n_spaces =` out of `if`.

Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/doc_test.ex]: Is it even possible to have `1` in this place?

Comment (eksperimental) [inline lib/ex_unit/lib/ex_unit/doc_test.ex]: fixed!
yes, it is possible. I have updated one of the tests to prove this

**Comment by eksperimental** [comment]
I would have added a test to check the message when it's expecting indenting with one space, but unless i want to modify an existing one (and i'ts not a good idea bc it's only checking messages of the most common cases with 2 and 4 spaces), I will have to correct the line numbers in the message and that is a real PITA!

File: lib/ex_unit/lib/ex_unit/doc_test.ex [hunk 2]
```diff
@@ -438,7 +438,11 @@ defmodule ExUnit.DocTest do
       ^stripped_line ->
         :ok
       _ ->
-        raise Error, message: "indentation level mismatch: #{inspect line}, should have been #{indent} spaces"
+        n_spaces = if indent == 1,
+          do: "#{indent} space",
+          else:  "#{indent} spaces"
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/doc_test.ex]: there is one extra space after `else:` (sorry for the nitpick)

File: lib/ex_unit/test/ex_unit/doc_test_test.exs [hunk 3]
```diff
@@ -12,8 +12,8 @@ defmodule ExUnit.DocTestTest.GoodModule do
   def test_fun, do: 1
 
   @doc ~S"""
-  iex> ~S(f#{o}o)
-  "f\#{o}o"
+   iex> ~S(f#{o}o)
```
Comment (josevalim) [inline lib/ex_unit/test/ex_unit/doc_test_test.exs]: There is no need to change this test, we are fine. :)

Comment (eksperimental) [inline lib/ex_unit/test/ex_unit/doc_test_test.exs]: yes, i realized of that after i submitted it ;-|

File: lib/ex_unit/lib/ex_unit/doc_test.ex (hunk 2 above)
Comment (eksperimental) [inline lib/ex_unit/lib/ex_unit/doc_test.ex]: no, worries. it's now or never

**Comment by eksperimental** [comment]
ok. all done

**Comment by eksperimental** [comment]
i have added some blank lines inside the case to make it clear

**Pull request merged**